frontend/src/pages/CreatePipeline/index.tsx

```diff
@@ -0,0 +1,686 @@
+            },
+        };
+        setNodesRef.current((nodes) => nodes.concat(newNode));
+    };
+
+    const checkInputsAreConnected = () : boolean => {
+        const exitNode = nodesRef.current.find((node) => node.type === "exitNodeEdit");
+        const lastEdge = edgesRef.current.find((edge) => edge.target === exitNode?.id);
+        if (!lastEdge) return false;
+        const allNodes = nodesRef.current.filter((node) => node.type !== "entryNodeEdit" && node.type !== "exitNodeEdit");
+        let inputConnected = true;
+        allNodes.forEach((node) => {
+            node.data.dataIn.forEach((input: FieldDescription) => {
+                if (!edgesRef.current.find((edge) => edge.targetHandle === input.name)) inputConnected = false;
+            })
+        })
+
+        return inputConnected;
+    }
+
+    const postPipeline = async () => {
+        if (!(await checkPipeline())) return;
+
+        const json = getJSONRepresentation();
+        createPipeline(json)
+            .then((answer) => {
+                if(answer?.errorBody) {
+                    toast(`Error: ${answer?.errorBody}`)
+                } else {
+                    toast("Pipeline created")
+                }
+            })
+            .catch(error => {
+                console.log("Error creating pipeline: ", error);
+            })
+    }
+
+
+    const checkPipeline = async () : Promise<boolean> => {
+
+        if (pipeName === '' || pipeSlug === '' || pipeSummary === '' || pipeDescription === '') {
+            toast("Please fill in the Pipeline information")
+            return false;
+        }
+
+        if (!checkInputsAreConnected()) {
+            toast("Invalid: Make sure every service input is connected")
+            return false;
+        }
+
+
+        try {
+            const json = getJSONRepresentation();
+            console.log(json);
+
+            const answer = await checkPipelineValidity(json);
+
+            if (answer && answer?.valid) {
+                toast("Pipeline is valid");
+                return true;
+            } else {
+                toast(`Invalid: ${answer?.errorBody}`);
+                return false;
+            }
+        } catch (error) {
+            toast("Could not check validity")
+            console.log("Error checking validity: ", error);
+            return false;
+        }
+
+    }
+
+    const getJSONRepresentation = (): string => {
+        const entryNode = nodesRef.current.find((node) => node.type === "entryNodeEdit");
+        const exitNode = nodesRef.current.find((node) => node.type === "exitNodeEdit");
+        const lastEdge = edgesRef.current.find((edge) => edge.target === exitNode?.id);
+
+        const data_in_fields = entryNode?.data.dataIn;
+        const data_out_fields = exitNode?.data.dataOut;
+        const queue: string[] = [lastEdge!.source]
+        let index = 0
+        while (index < queue.length) {
+            const node = nodesRef.current.find((nd) => nd.id === queue[index])
```

Comment:
ESLint: Function declared in a loop contains unsafe references to variable(s) 'index'.(no-loop-func)